aws/resource_aws_storagegateway_fsx_associate_file_system.go

```diff
@@ -0,0 +1,304 @@
+	conn := meta.(*AWSClient).storagegatewayconn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	input := &storagegateway.DescribeFileSystemAssociationsInput{
+		FileSystemAssociationARNList: []*string{aws.String(d.Id())},
+	}
+
+	log.Printf("[DEBUG] Reading Storage Gateway FSx File Systems: %s", input)
+	output, err := conn.DescribeFileSystemAssociations(input)
+	if err != nil {
+		if tfawserr.ErrCodeEquals(err, storagegateway.ErrCodeInvalidGatewayRequestException) {
+			var igrex *storagegateway.InvalidGatewayRequestException
+			if ok := errors.As(err, &igrex); ok {
+				if err := igrex.Error_; err != nil {
+					if aws.StringValue(err.ErrorCode) == "FileSystemAssociationNotFound" {
+						log.Printf("[WARN] FSX File System %q not found, removing from state", d.Id())
+						d.SetId("")
+						return nil
+					}
+				}
+			}
+		}
+
+		return fmt.Errorf("error reading Storage Gateway FSx File System: %w", err)
+	}
+
+	if output == nil || len(output.FileSystemAssociationInfoList) == 0 || output.FileSystemAssociationInfoList[0] == nil {
+		log.Printf("[WARN] Storage Gateway FSx File System %q not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	filesystem := output.FileSystemAssociationInfoList[0]
+
+	arn := filesystem.FileSystemAssociationARN
+	d.Set("arn", arn)
+
+	d.Set("audit_destination_arn", filesystem.AuditDestinationARN)
+	d.Set("gateway_arn", filesystem.GatewayARN)
+	d.Set("location_arn", filesystem.LocationARN)
+
+	if err := d.Set("cache_attributes", flattenStorageGatewayFsxAssociateFileSystemCacheAttributes(filesystem.CacheAttributes)); err != nil {
+		return fmt.Errorf("error setting cache_attributes: %w", err)
+	}
+
+	tags := keyvaluetags.StoragegatewayKeyValueTags(filesystem.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig)
+
+	//lintignore:AWSR002
+	if err := d.Set("tags", tags.RemoveDefaultConfig(defaultTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	if err := d.Set("tags_all", tags.Map()); err != nil {
+		return fmt.Errorf("error setting tags_all: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAwsStorageGatewayFsxAssociateFileSystemUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).storagegatewayconn
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+		if err := keyvaluetags.StoragegatewayUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating tags: %w", err)
+		}
+	}
+
+	if d.HasChanges("cache_attributes", "audit_destination_arn", "username", "password") {
+
+		input := &storagegateway.UpdateFileSystemAssociationInput{
+			Password:                 aws.String(d.Get("password").(string)),
+			UserName:                 aws.String(d.Get("username").(string)),
+			FileSystemAssociationARN: aws.String(d.Id()),
+		}
+
+		if v, ok := d.GetOk("audit_destination_arn"); ok {
```

Comment:
as discussed. i think that disabling logs is skipped because GetOk might be skipping `""` lets try testing. with  putting `AuditDestinationARN` in the above block as it defaults always to `""`

Reply:
Updated.